src/routes/index.js

```diff
@@ -1,28 +1,244 @@
-const schema = {
-  response: {
-    200: {
-      type: 'object',
-      properties: {
-        message: { type: 'string' },
-        data: { type: 'object' },
-      },
-    },
-  },
-}
+const { signToken } = require('../helpers/signToken')
+const { decipherToken } = require('../helpers/decipherToken')
+const asyncHandler = require('../middlewares/asyncHandler')
+const ErrorResponse = require('../managers/error/ErrorResponse')
+const User = require('../entities/User')
+const { statusCodes, codes } = require('../managers/error/constants')
+const Password = require('../helpers/password')
+const { v4: uuidv4 } = require('uuid')
+const ChatMessage = require('../entities/ChatMessage')
+
+const routes = async (fastify) => {
+  // Sign up
+  fastify.post(
+    '/signup',
+    asyncHandler(async (request, reply) => {
+      const { username, password } = request.body
+
+      if (!username || username === '') {
+        throw new ErrorResponse({
+          title: 'Please provide an username.',
+          code: codes.ERR_PARAMETER_REQUIRED,
+          status: statusCodes.UNPROCESSABLE_ENTITY,
+          pointer: 'username',
+        })
+      } else if (!password || password === '') {
+        throw new ErrorResponse({
+          title: 'Password is required.',
+          code: codes.ERR_PARAMETER_REQUIRED,
+          status: statusCodes.UNPROCESSABLE_ENTITY,
+          pointer: 'password',
+        })
+      }
+
+      // TODO: Add unique constraint for username
+
+      const id = uuidv4()
+
+      await User.put({
+        id,
+        username,
+        password,
+      })
+
+      reply.send({
+        token: signToken(id),
+      })
+    })
+  )
+
+  // Sign in
+  fastify.post(
+    '/signin',
+    asyncHandler(async (request, reply) => {
+      const { username, password } = request.body
+
+      if (!username || username === '') {
+        throw new ErrorResponse({
+          title: 'Please provide an username.',
+          code: codes.ERR_PARAMETER_REQUIRED,
+          status: statusCodes.UNPROCESSABLE_ENTITY,
+          pointer: 'username',
+        })
+      } else if (!password || password === '') {
```

Comment:
same.  `''` should be covered in `!password`, since empty string evaluates to false.